core-backend/sandbox/src/funcs.rs

```diff
@@ -126,17 +140,21 @@ where
         let (pid_value_ptr, payload_ptr, len, delay, err_mid_ptr) = args.iter().read_5()?;
 
         ctx.run(|ctx| {
+            let read_hash_val = ctx.register_read_as(pid_value_ptr);
+            let read_payload = ctx.register_read(payload_ptr, len);
+            let write_err_mid = ctx.register_write_as(err_mid_ptr);
```

Comment:
do we expect that once we register write this will be cleaned at caller on first read?

Reply:
sent details in dm

Reply:
> do we expect that once we register write this will be cleaned at caller on first read?

Yes all registered accesses will be pre-processed when the first one access is called.

Reply:
Will add TODO in next PR